### ISSUE TITLE
Incorrect splitting of stored procedure

### ISSUE THREAD

**Issue by fbertsch** [andialbrecht/sqlparse#583]
For a specific stored procedure, I've found that the splitting splits early.

Here is a reproducible example. I've bisected a bit to identify the `CONCAT` as the culprit:
```
import sqlparse

fails_parse = '''
CREATE OR REPLACE PROCEDURE
  event_analysis.get_funnel_steps_sql(funnel ARRAY<STRUCT<step_name STRING, list ARRAY<STRUCT<category STRING, event_name STRING>>>>)
BEGIN
   DECLARE i INT64 DEFAULT 1;
   WHILE i <= ARRAY_LENGTH(funnel) DO
     SET funnel_sqls = ARRAY_CONCAT(funnel_sqls, [CONCAT('r', funnel_step_regex[ORDINAL(i)])]);
     SET i = i + 1;
  END WHILE;
END;
'''

succeeds_parse = '''
CREATE OR REPLACE PROCEDURE
  event_analysis.get_funnel_steps_sql(funnel ARRAY<STRUCT<step_name STRING, list ARRAY<STRUCT<category STRING, event_name STRING>>>>)
BEGIN
  DECLARE i INT64 DEFAULT 1;
  WHILE i <= ARRAY_LENGTH(funnel) DO
    SET funnel_sqls = ARRAY_CONCAT(funnel_sqls, [funnel_step_regex[ORDINAL(i)]]);
     SET i = i + 1;
   END WHILE;
 END;
'''

assert(len(sqlparse.split(fails_parse)) == 3)
assert(len(sqlparse.split(succeeds_parse)) == 2)
```

**Comment by andialbrecht** [comment]
Thanks for reporting this issue! It has been fixed and will be available in the soon to be released 0.4.0 version.

**Comment by fbertsch** [comment]
@andialbrecht wow, thank you for the quick response and resolution! I really appreciate you taking the time to fix it.
